src/vcModals.cpp

```diff
@@ -340,51 +315,23 @@ void vcModals_DrawFileModal(vcState *pProgramState)
 
     ImGui::Separator();
 
-    if (mode == vcMT_AddUDS)
+    if (mode == vcMT_AddSceneItem)
     {
-      const char *fileExtensions[] = { ".uds", ".ssf", ".udg" };
-      if (vcFileDialog_Show(pProgramState->modelPath, sizeof(pProgramState->modelPath), true, fileExtensions, udLengthOf(fileExtensions)))
-        loadFile = true;
-    }
-    else if (mode == vcMT_ImportProject)
-    {
-      const char *fileExtensions[] = { ".json", ".udp" };
-      if (vcFileDialog_Show(pProgramState->modelPath, sizeof(pProgramState->modelPath), true, fileExtensions, udLengthOf(fileExtensions)))
-        loadFile = true;
+      vcIGSW_FilePicker(pProgramState, "Filename", pProgramState->modelPath, SupportedFileTypes_SceneItems, true, [pProgramState] {
+        pProgramState->loadList.PushBack(udStrdup(pProgramState->modelPath));
+      });
     }
     else if (mode == vcMT_ExportProject)
     {
       const char *fileExtensions[] = { ".json" };
-      if (vcFileDialog_Show(pProgramState->modelPath, sizeof(pProgramState->modelPath), false, fileExtensions, udLengthOf(fileExtensions)))
+      if (vcFileDialog_DrawImGui(pProgramState->modelPath, sizeof(pProgramState->modelPath), false, fileExtensions, udLengthOf(fileExtensions)))
```

Comment:
Isn't there a template helper to avoid these `sizeof` and `udLengthOf` calls?

Reply:
There is not.